src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -2595,6 +2595,12 @@ public static class Extensions
                 // (8,9): error CS1510: A ref or out value must be an assignable variable
                 //         (global, global) = global;
                 Diagnostic(ErrorCode.ERR_RefLvalueExpected, "(global, global) = global").WithLocation(8, 9),
+                // (8,9): error CS8352: Cannot use local '(global, global) = global' in this context because it may expose referenced variables outside of their declaration scope
+                //         (global, global) = global;
+                Diagnostic(ErrorCode.ERR_EscapeLocal, "(global, global) = global").WithArguments("(global, global) = global").WithLocation(8, 9),
+                // (8,28): error CS8350: This combination of arguments to 'Extensions.Deconstruct(ref Span<int>, out Span<int>, out Span<int>)' is disallowed because it may expose variables referenced by parameter 'x' outside of their declaration scope
+                //         (global, global) = global;
+                Diagnostic(ErrorCode.ERR_CallArgMixing, "global").WithArguments("Extensions.Deconstruct(ref System.Span<int>, out System.Span<int>, out System.Span<int>)", "x").WithLocation(8, 28),
```

Comment:
Are these additional diagnostics expected?

Reply:
We were losing ref-ness here, so the conditions for these errors were not even detected. 
Considering this is an error scenario overall, I am not sure if these errors should be reported, but they were not reported originally for a wrong reason - we were not detecting them at all. 
Now the errors are produced as a result of the binding. Whether these errors should be reported or dropped is a separate question and not a part of this fix IMO.

---
In reply to: [160800083](https://github.com/dotnet/roslyn/pull/24128#discussion_r160800083) [](ancestors = 160800083)